File: scheduler/core/src/main/java/io/orkes/conductor/dao/archive/SchedulerArchivalDAO.java
package io.orkes.conductor.dao.archive;

import java.util.List;
import java.util.Map;
import java.util.Set;

import com.netflix.conductor.common.run.SearchResult;

import io.orkes.conductor.scheduler.model.WorkflowScheduleExecutionModel;

public interface SchedulerArchivalDAO {
    void saveExecutionRecord(WorkflowScheduleExecutionModel executionModel);

    SearchResult<String> searchScheduledExecutions(
            String query, String freeText, int start, int count, List<String> sort);

    Map<String, WorkflowScheduleExecutionModel> getExecutionsByIds(Set<String> executionIds);

    WorkflowScheduleExecutionModel getExecutionById(String executionId);

    void cleanupOldRecords(int archivalMaxRecords, int archivalMaxRecordThreshold);
}

File: scheduler/core/src/main/typescript/io/orkes/conductor/dao/archive/SchedulerArchivalDAO.ts
import { SearchResult } from 'com/netflix/conductor/common/run/SearchResult';
import { WorkflowScheduleExecutionModel } from 'io/orkes/conductor/scheduler/model/WorkflowScheduleExecutionModel';

export interface SchedulerArchivalDAO {
    saveExecutionRecord(executionModel: WorkflowScheduleExecutionModel): void;

    searchScheduledExecutions(
        query: string,
        freeText: string,
        start: number,
        count: number,
        sort: Array<string>
    ): SearchResult<string>;

    getExecutionsByIds(executionIds: Set<string>): Map<string, WorkflowScheduleExecutionModel>;

    getExecutionById(executionId: string): WorkflowScheduleExecutionModel | null;

    cleanupOldRecords(archivalMaxRecords: number, archivalMaxRecordThreshold: number): void;
}
